sql/api/src/main/scala/org/apache/spark/sql/types/StringType.scala

```diff
@@ -26,7 +27,27 @@ import org.apache.spark.annotation.Stable
  * @param collationId The id of collation for this StringType.
  */
 @Stable
-class StringType private(val collationId: Int) extends AtomicType {
+class StringType private(val collationId: Int) extends AtomicType with Serializable {
+  /**
+   * Returns whether assigned collation is the default spark collation (UCS_BASIC).
+   */
+  private def isDefaultCollation: Boolean = collationId == StringType.DEFAULT_COLLATION_ID
```

Comment:
Why not making this public? It might be useful in other places.

Reply:
Yeah, it is public now. Just didn't have a use case for this method prior to latest changes so I kept it private.